### DISCUSSION TITLE
More consistent language themes

### DISCUSSION THREAD

**Discussion by jtlandis** [zed-industries/zed#12550] [Feature Requests]
Is there a way to customize coding language themes through tree-sitter?
I know extensions exist to [change the overall theme of the Zed app](https://zed.dev/blog/user-themes-now-in-preview), and extensions to add tree-sitter for other coding languages exist.

I ask this question because it seems that there isn't a standard for how syntax highlighting schemes are specified. For instance, a scheme file could map queries to some arbitrary name.

This can leave some coding languages feeling a bit dull depending on how the theme styling specified.

It would be nice if themes were to export a palette, so that users can specify a certain palette index in `experimental.theme_override` for the arbitrary remap found in `.scm`. 

This may make switching themes more enjoyable.

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
